camayoc/tests/qpc/api/v1/reports/test_reports.py

```diff
@@ -151,7 +151,7 @@ def test_OS_found_deployment_report(scans, scan_name):
             found_version = ""
         found_is_redhat = actual_data.get("is_redhat", SENTINEL)
 
-        if expected_distribution.release not in found_release:
+        if found_release.startswith(expected_distribution.release) is False:
```

Comment:
blocker:
```suggestion
        if not found_release.startswith(expected_distribution.release):
```

I would expect one of static checkers to shout about it, PEP 8 explicitly calls out comparing booleans.

Same applies to line 167 below.

Reply:
Interesting discussion. This coding style is said to be wrong in PEP 8, but I don't see around a hard enforcement against for testing a condition with `is` with True or False, there are uses of this style in:

* Python library  https://github.com/python/cpython/blob/main/Lib/multiprocessing/sharedctypes.py#L75
* Django https://github.com/django/django/blob/main/django/core/checks/security/sessions.py#L68

My understanding is testing implicity for boolean goes against the idea of _Explicit is better than implicit_ in PEP 20.

I know this PR case is trivial and your suggestion is shorter and equivalent, but it is safe to test explicitly with _truthy_ objects in general. I think requests exemplificates https://github.com/psf/requests/blob/main/src/requests/adapters.py#L111

I don't believe this is a bocker.

Reply:
This is an interesting discussion. The only time I'd not follow PEP 8 to verify a bool with `is` (especially `is False`) is when that variable can be set to `None`, `0`, `""`, etc. (any value that would eval `not variable` as `True`) IF it is possible to have that value set to something other than bool AND we need a specific behavior only when it is a bool.

Otherwise, go with `not some_var`, it is more readable.

Regardless, I agree with you @ruda : this isn't a blocker.

Reply:
Generally I would expect linter to pick this up, hence my surprise it did not.
PEP 8 explicitly calls that specific syntax as "worse" [link](https://peps.python.org/pep-0008/#programming-recommendations):

![Screenshot from 2024-11-13 09-52-18](https://github.com/user-attachments/assets/18116b9e-e0d6-4025-8da6-8c9577973a98)

"Blocker" was bad phrasing on my part. I posted two comments and my intent was to clarify which one contributes to "request changes" decision.

Let’s leave it as it is. I should have not brought up styling issues when tools did not complain.